### DISCUSSION TITLE
Create own framework

### DISCUSSION THREAD

**Discussion by Jak2k** [vikejs/vike#1074] [Discussion]
### Description

The ability to create an own framework with vite-plugin-ssr sounds very cool. But I have a few questions:

- Do I need to include special things in the framework package (except from the render folder)?
- How do I need to register the package?
- Is it possible to include vite and vite-plugin-ssr in that package and then just redirect script calls like `dev`,`build` so that users just need to install my package?

**Comment by brillout** [comment]
What is it you want to build? A company internal framework? An open source framework?

(Moving to discussions as writing elaborate docs for this isn't a goal in the short/middle term.)

**Reply by Jak2k** [reply]
I would like to create an open source framework for building apps with vanilla css.

**Reply by brillout** [reply]
The best documentation out there are the examples https://github.com/brillout/restack and https://github.com/brillout/solide.

At this point it's best to have a a look at these and see how it all works.

**Reply by Jak2k** [reply]
Where is the related source code in VPS? Maybe reading it could also help understanding it.

**Reply by brillout** [reply]
There isn't really a related source code to it. Considering the questions you asked, I think the best way for you to get started is to first get familiar with VPS and then, once you've got a feeling for how VPS works, check out these two examples.
